ouroboros-consensus-byron-test/test/Test/ThreadNet/RealPBFT.hs

```diff
@@ -116,14 +153,18 @@ tests = testGroup "RealPBFT" $
       testProperty "addressed by InvalidRollForward exception (PR #773)" $
           once $
           let ncn = NumCoreNodes 3 in
-          prop_simple_real_pbft_convergence ProduceEBBs (SecurityParam 10) TestConfig
-            { numCoreNodes = ncn
-            , numSlots     = NumSlots 24
-            , nodeJoinPlan = NodeJoinPlan $ Map.fromList [(CoreNodeId 0,SlotNo 0), (CoreNodeId 1,SlotNo 20), (CoreNodeId 2,SlotNo 22)]
-            , nodeRestarts = noRestarts
-            , nodeTopology = meshNodeTopology ncn
-            , slotLength   = defaultSlotLength
-            , initSeed     = Seed (15069526818753326002, 9758937467355895013, 16548925776947010688, 13173070736975126721, 13719483751339084974)
+          prop_simple_real_pbft_convergence TestSetup
+            { setupEBBs       = ProduceEBBs
+            , setupK          = SecurityParam 10
+            , setupTestConfig = TestConfig
+              { initSeed     = Seed (15069526818753326002, 9758937467355895013, 16548925776947010688, 13173070736975126721, 13719483751339084974)
```

Comment:
Just a thought for when we do redesign: we should also try hard to get rid of this `seed` parameter. It's inherently shrinkable, and very brittle.

Reply:
Hmm. I need to think more what it would look like to replace it. My hunch is that we can't (without a lot of work), since the nodes have to generate eg transactions that depend on behaviors of the system under test that we currently can't always anticipate (eg Praos leader schedules, which leaders non-leaders follow, etc).

For the Praos tests I'm developing with message delays, the "statistical test" aspects from the QuviQ presentation use an "internal" `forAll` in order to generate enough samples to satisfy the statistical test's desired confidence level. That `forAll` seems like the right place for generating this kind of seed. Not sure how to generalize that observation to the rest of the protocols, especially the ones that I haven't been anticipating using statistical testing for.

Reply:
One thing that John emphasizes a _lot_ is that the test we run _should not depend on the execution of that test_. Everything that we're going to throw at the test should be determined _a priori_, so that it is inspectable, reproducible, and crucially, shrinkable. Of course, achieving this for the kinds of tests that we want to do is not easy, but we should strive hard hard to get there.